src/node.cc

```diff
@@ -165,23 +168,21 @@ ArrayBufferAllocator ArrayBufferAllocator::the_singleton;
 
 
 void* ArrayBufferAllocator::Allocate(size_t length) {
-  if (length > kMaxLength)
+  void* data = malloc(length);
+  if (data == nullptr)
     return nullptr;
-  char* data = new char[length];
   memset(data, 0, length);
   return data;
 }
```

Comment:
Will do. Didn't know of there was some reason unknown to me for doing it the current way (which is the same way it's done in d8).